proposals/005_New_Security_Definitions.md

```diff
@@ -0,0 +1,531 @@
+* Update `type` to allow for any value instead of the hard-coded list of supported types
+
+## Detailed design
+
+The bulk of this proposal is defining a way to separate the properties used to describe the credential(s) for a Security
+Scheme from the root properties of the Security Scheme.  This proposal will also allow you to define multiple
+credentials _(multiple locations, formats, ...)_ for the same Security Scheme for cases where you might support
+multiple formats/locations for effectively the same credential.  The new objects being suggested as part of this
+proposal are defined below.
+
+**Note:** While not specifically mentioned below for brevity, JSON References and Extensions should be allowed within
+every complex type documented below.
+
+### Value Getter Object
+
+The _Value Getter Object_ describes a place in the request where something exists that you want to retrieve.  The name
+and purpose of this object is generic, as in I could see other places in OpenAPI where you could point to a value in the
+request.  The _Value Getter Object_ is the base object to be _extended_ by **all**
+[Credential Object](#credential-object)s.  Below is its schema:
+
+| Field Name | Type | Applies To | Description |
+| ---------- | ---- | ---------- | ----------- |
+| in | `string` | Any | **REQUIRED.** Specifies where in the request the credential is provided.  Valid values are: `body`, `cookie`, `header`, `path` and `query` |
+| format | `regex` | Any | Specifies the format of the credential.  Since in many cases, the credential is a subset of the raw value, `format` must be a regular expression that matches the complete value, and contains a single capturing group to match the actual value within the raw value. |
+| name | `string` | Any | **REQUIRED.** Specifies the name used to resolve the credential in the location provided by `in`. |
+
+**Note:** For `body`, `name` should be a JSON Pointer _(or something similar)_ that lets you define what part of the
+body you are interested in.  For example, you could use `#` to dictate that body in its entirety is the credential or
+you could use something like `#/some/path` to dictate that a portion of the body is the credential.  I understand that
+JSON Pointer usage implies that the body is JSON but that doesn't have to be the case and could be used to address any
+structured data.
+
+**Note:** For `format`, I realize that it being a regular expression might seem a bit offputting but there is no way to
+reasonably support raw values _(as-is)_ and taking a raw value and only grabbing the important part
+_(like when using HTTP Auth where you don't want the scheme details)_.  We could also have _named formats_ that would
+alleviate the need for consumers to author regular expressions themselves for things like HTTP Auth, Bearer Tokens, etc.
+
+**Note:** I could see a case for a `description` field.
+
+**Note:** I could see a case for supporting an `in` for mTLS
```

Comment:
How would the certificate be presented in a `query` parameter, `header`, `cookie` or `body`? Unless presented as part of a TLS handshake, you're not doing mutual TLS.

Reply:
I get what you mean.  mTLS is done at the protocol level so actually providing the certificate as part of the request wouldn't be useful.  I'll remove this.